samples/connectivity/callnotification/src/main/java/com/example/platform/connectivity/callnotification/CallNotificationSample.kt

```diff
@@ -0,0 +1,191 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+import com.google.android.catalog.framework.annotations.Sample
+
+@Sample(
+    name = "Call Notification Sample",
+    description = "Sample demonstrating how to make incoming call notifications and in call notifications",
+)
+class CallNotificationSample : ComponentActivity() {
+
+    var notificationSource: NotificationSource<NotificationReceiver>? = null
+
+    @OptIn(ExperimentalPermissionsApi::class)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        notificationSource = NotificationSource(this, NotificationReceiver::class.java)
+
+        setContent {
+            MaterialTheme {
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = colorScheme.background,
+                ) {
+                    // We should be using make_own_call permissions but this requires
+                    // implementation of the telecom API to work correctly.
+                    // Please see telecom example for full implementation
+                    val callPermission = rememberPermissionState(
+                        Manifest.permission.POST_NOTIFICATIONS,
+                    )
+
+                    if (callPermission.status.isGranted) {
+                        EntryPoint(notificationSource!!)
+                    } else {
+                        PermissionWidget(callPermission)
+                    }
+                }
+            }
+        }
+    }
+
+    class NotificationReceiver : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            intent?.let {
+                val notificationStateValue = it.getIntExtra(
+                    NotificationSource.NOTIFICATION_ACTION,
+                    NotificationSource.Companion.NotificationState.CANCEL.ordinal,
+                )
+
+                var message = "No User Input"
+
+                when (notificationStateValue) {
+                    NotificationSource.Companion.NotificationState.ANSWER.ordinal -> message =
+                        "Answered"
+
+                    NotificationSource.Companion.NotificationState.REJECT.ordinal -> message =
+                        "Rejected"
+
+                    else -> {
+                        message =
+                            "Cancelled"
+                    }
+                }
+
+                Toast.makeText(context, message, Toast.LENGTH_LONG).show()
```

Comment:
we should be steering away from Toast in favor of snackbars or a notification. 

It is captured in DAC though someone has to proactively sweep projects and DAC to further this effort.
https://developer.android.com/develop/ui/views/notifications/snackbar

Reply:
Agreed, but in this case its alot simpler to just setup a toast message.
Larger projects yes lets use snackbars.

Reply:
Can you comment in-code a that for just the purpose of this project we make use of toast but otherwise pls use the alternatives.